select.js

```diff
@@ -1,14 +1,39 @@
-define(function() {
-    var init = function() {
-        $('body').on('change', '.c-select', function() {
-            var $container = $(this);
-            var value = $container.find('option:selected').text();
-
-            $container.find('.c-select__value').text(value);
-        });
-    };
+define(['$'], function($) {
+    //
+    // Component constructor
+    //
+    var Select = function Select($el, options) {
+        // Component instance properties. The `$` hash (by convention) caches
+        // the root node and any sub-components.
+        this.value = '';
+        this.dom = {
+            $root: $el,
+            $value: $el.find('.c-select__value')
+        };
+
+        // Event handlers should use $.proxy to preserve access to the instance
+        // as `this` within the handler.
+        this.dom.$root.on('change', $.proxy(this.update, this));
+
+        // Immediately invoke any methods needed to set up initial state.
+        this.update();
+    }
+
+    //
+    // Update the component’s visible value when the underlying select element’s
+    // value changes.
+    //
+    Select.prototype.update = function update() {
+        this.value = this.dom.$root.find('option:selected').text();
+        this.dom.$value.text(this.value);
+        this.dom.$root.trigger('select:update'); // notify subscribers
+    }
 
     return {
-        'init': init
+        'init': function($el, options) {
+            // If already initialized, return the instance; otherwise, create it
+            // and expose it through `$('.c-select').data('component')`.
+            return $el.data('component') || $el.data('component', new Select($el, options));
```

Comment:
I brought this up with @ry5n in chat, but we realized that in jQuery `.data()` behaves differently than in default vanilla Zepto. Thoughts on whether this is a concern? This is a question because it sounds like we do this already in plugins, for example...

```
!plugin && $this.data(name, (plugin = new SubConstructor(this, option)));
```